docs/source/programming/GIN-repositories.md

```diff
@@ -0,0 +1,338 @@
+       ```
+
+       where `<remote-name>` is the name you want to give to the remote (e.g. `origin`) and `<remote-repository-location>` is the location of the data store, which should be in the form of alias:path or server:path (e.g. `gin add-remote origin gin:sfmig/crab-data`).
+
+       - If the remote GIN repository doesn't exist, you will be prompted to either create the remote GIN repository, add the remote address anyways, or abort.
+       - To show the remotes accessible to your GIN account run `gin remotes`.
+       </details>
+
+:::{note}
+Initialising the GIN local repository (with `gin create` or `gin init`) will create a hidden `.git` subdirectory. You can see it on the terminal by running `ls -la` from the local repository. The local repository excluding this `.git` folder is what we call later the _working directory_.
+:::
+
+:::{tip}
+To create a GIN repository on a `ceph` directory:
+
+- You may need to mount the `ceph` directory first. To do this temporarily (i.e., until the next reboot), follow [this guide](https://howto.neuroinformatics.dev/programming/Mount-ceph-ubuntu-temp.html). To do this permanently, follow [this one](https://howto.neuroinformatics.dev/programming/Mount-ceph-ubuntu.html).
+- You may also need to add an exception for the mounted directory. To do so, run the following command:
+
+   ```
+   git config --global --add safe.directory /mnt/<path-to-the-mounted-directory
+   ```
+
+ - Alternatively, you can log to SWC's HPC cluster (specifically, its [gateway node](https://howto.neuroinformatics.dev/_images/swc_hpc_access_flowchart.png) `hpc-gw1`), which has the GIN CLI client installed, and work from there. This is likely faster than mounting the `ceph` directory in your laptop, since the HPC cluster is in the same network as `ceph` (and actually physically close to it).
+:::
+
+3. **Add files to the GIN remote repository**
+
+   It is good practice to keep a record of the changes in the repository through commit messages. To keep a useful and clean commit history, it is also recommended to make small commits by selecting a subset of the files.
+
+   - To make a record of the current state of a local repository, run
+
+     ```
+     gin commit --message <message> <filename>
+     ```
+
+     You can replace the `filename` above by an expression with wildcard (e.g., `*.png` to include all png files). It can also be a list of files (separated by white spaces). A filename equal to `.` will include all files with changes. See the full syntax [here](https://gin.g-node.org/G-Node/Info/wiki/GIN+CLI+Help#record-changes-in-local-repository).
+
+   - To upload all local changes to the remote GIN repository:
+
+     ```
+     gin upload <filename>
+     ```
+
+     `filename` accepts the same inputs as in `gin commit`. Again, the recommended practice would be to upload data in small-ish chunks. You can run an upload command after a few commits (so not necessarily after each commit).
```

Comment:
I am not sure how to estimate this - my experience was that if I uploaded many (~hundreds?) files (images of ~12MB) at once it would more likely fail.

I removed it for now but let me know if you'd like a better phrasing / estimate.